python/packages/aws-lambda-sdk/pyproject.toml

```diff
@@ -0,0 +1,29 @@
+[build-system]
+build-backend = "setuptools.build_meta"
+requires = [
+    "setuptools>=65.6.3",
+    "wheel>=0.38.4",
+]
+
+[project]
+name = "serverless-aws-lambda-sdk"
+version = "0.0.0"
+description = "Serverless AWS Lambda SDK for Python"
+authors = [{ name = "serverlessinc" }]
+requires-python = ">=3.7"
+dependencies = [
+    "serverless-sdk~=0.1",
+    "serverless-sdk-schema~=0.1",
+    "strenum>=0.4.9",
+    "typing-extensions>=4.4", # included in Python 3.8 - 3.11
```

Comment:
Shouldn't we here also rely on `~` ? Are those package following semver? How can we prevent automatic upgrades in case of breaking changes?

Reply:
It's not clear whether these dependencies follow semver. I've amended it to make sure we lock the minor version of our dependencies.

I can't think of a way to 100% prevent downstream dependencies to break our packages, because we don't control those packages.

Reply:
> I can't think of a way to 100% prevent downstream dependencies to break our packages, because we don't control those packages.

In that case, key is understanding the version scheme they're following. If they follow semver then we know the rules, otherwise, each patch can be breaking